clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1181,15 +1181,18 @@ def ShiftOp : CIR_Op<"shift", [Pure]> {
   let summary = "Shift";
   let description = [{
     Shift `left` or `right`, according to the first operand. Second operand is
-    the shift target and the third the amount.
+    the shift target and the third the amount. Second and the thrid operand can
+    be either integer type or vector of integer type. However, they must be
+    either all vector of integer type, or all integer type.  
```

Comment:
Can you expand a bit on the semantics? I assume it shifts all values in each of the lanes by the same amount?

Reply:
will do, and yes, that is the intended semantic is that all elements in shift amount vector should be the same.